File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.cpp
/**
 * Definition for a binary tree node.
 * struct TreeNode {
 *     int val;
 *     TreeNode *left;
 *     TreeNode *right;
 *     TreeNode() : val(0), left(nullptr), right(nullptr) {}
 *     TreeNode(int x) : val(x), left(nullptr), right(nullptr) {}
 *     TreeNode(int x, TreeNode *left, TreeNode *right) : val(x), left(left), right(right) {}
 * };
 */
class Solution {
public:
    vector<int>ans;
    void helper(TreeNode* root, int level, int& mxlevel){
        if(root==NULL){
            return;
        }
        if(level>mxlevel){
            ans.push_back(root->val);
            mxlevel=level;
        }
        helper(root->right,level+1,mxlevel);
        helper(root->left,level+1,mxlevel);
    }
    vector<int> rightSideView(TreeNode* root) {
        int mxlevel=-1;
        helper(root,0,mxlevel);
        return ans;
    }
};

File: 0199-binary-tree-right-side-view/0199-binary-tree-right-side-view.java
/**
 * Definition for a binary tree node.
 * public class TreeNode {
 *     int val;
 *     TreeNode left;
 *     TreeNode right;
 *     TreeNode() {}
 *     TreeNode(int val) { this.val = val; }
 *     TreeNode(int val, TreeNode left, TreeNode right) {
 *         this.val = val;
 *         this.left = left;
 *         this.right = right;
 *     }
 * }
 */
class Solution {
    public List<Integer> rightSideView(TreeNode root) {
        List<Integer> ans=new ArrayList<>();
        if(root==null){
            return ans;
        }

        Queue<TreeNode>q=new LinkedList<>();
        q.offer(root);

        while(!q.isEmpty()){
            int sz=q.size();

            for(int i=0;i<sz;i++){
                TreeNode curr=q.poll();
                if(i==sz-1){
                    ans.add(curr.val);
                }

                if(curr.left!=null){
                    q.offer(curr.left);
                }
                if(curr.right!=null){
                    q.offer(curr.right);
                }
            }
        }
        return ans;
    }
}